docs/user-guide/rug/event-handlers.md

```diff
@@ -0,0 +1,204 @@
+connected and acted upon provides tremendous value for shipping better code
+faster. The mechanism for realizing this promise is to automate event responses
+using Rug event handlers.
+
+Event handlers define automated responses to events. Each event handler is a
+program in a compiles-to-JavaScript, Turing-complete language. Based on the
+characteristics of the event that occurred, the handler decides whether to act and
+which actions to take.
+
+!!! hint "Why not try..."
+    A new issue was created? Post that in the repository's chat channel and also add
+    buttons to the message that let people apply labels or claim the issue without
+    leaving chat.
+
+    A developer submits a pull request in a library? Find out whether it will to
+    impact a service that uses the library: create a branch in the service, modify
+    the code to update the dependency.
+
+    The service build completes successfully?
+    Update the library's pull request (PR), and tell the developer all the news.
+
+    A person in chat asks Atomist "what did I do today?"? Respond by
+    listing the issues they updated, the PRs they reviewed, and the commits they
+    pushed.
+
+## Anatomy of an Event Handler
+
+Rug event handlers are where we respond to the events that matter in your system.
+
+Depending on their goal, Rug event handler implementations are stored either along
+side the project they target or in different projects altogether if they are generic. 
+In both cases however, Rug event handlers share the same approach as other Rugs. For the 
+purpose of this documentation, we will assume our handler implementations live in 
+their own projects rather than in an existing project as the documentation describes
+for Rug generators or editors.
+
+{!handler-layout.md!}
+
+### Example Event Handler
+
+The following event handler responds when a tag is added to or deleted from an
+issue. When this event occurs, the Atomist Bot sends a message to the
+repository's Slack channel informing people in the channel about the tagging
+_event_.
+
+```typescript linenums="1"
+import { EventHandler, Tags } from "@atomist/rug/operations/Decorators";
+import { HandleEvent, LifecycleMessage, Plan } from "@atomist/rug/operations/Handlers";
+import { GraphNode, Match, PathExpression } from "@atomist/rug/tree/PathExpression";
+
+import { Comment } from "@atomist/cortex/Comment";
+import { Issue } from "@atomist/cortex/Issue";
+
+@EventHandler("ClosedGitHubIssues", "Handles closed issue events",
+    new PathExpression<Issue, Issue>(
+        `/Issue()[@state='closed']
+            [/resolvingCommits::Commit()/author::GitHubId()
+                [/person::Person()/chatId::ChatId()]?]?
+            [/openedBy::GitHubId()
+                [/person::Person()/chatId::ChatId()]?]
+            [/closedBy::GitHubId()
+                [/person::Person()/chatId::ChatId()]?]?
+            [/repo::Repo()/channels::ChatChannel()]
+            [/labels::Label()]?`))
```

Comment:
Wow, we may want to ease them into path expressions rather than accost them with this monstrosity right off the bat.